apps/SageAccountWeb/src/components/ResetTwoFactorAuth.tsx

```diff
@@ -0,0 +1,164 @@
+import { Alert, Box, Button, Typography } from '@mui/material'
+import React, { useMemo, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import {
+  displayToast,
+  StandaloneLoginForm,
+  SynapseQueries,
+  useApplicationSessionContext,
+} from 'synapse-react-client'
+import {
+  TwoFactorAuthErrorResponse,
+  TwoFactorAuthResetToken,
+} from '@sage-bionetworks/synapse-types'
+import { getSearchParam, hexDecodeAndDeserialize } from '../URLUtils'
+import { BackButton } from './BackButton'
+import { LeftRightPanel } from './LeftRightPanel'
+import { SourceAppLogo } from './SourceApp'
+import { RESET_2FA_SIGNED_TOKEN_PARAM } from '../Constants'
+
+export function ResetTwoFactorAuth() {
+  const history = useHistory()
+  const { refreshSession, twoFactorAuthSSOErrorResponse } =
+    useApplicationSessionContext()
+
+  const twoFactorAuthResetTokenParam = getSearchParam(
+    RESET_2FA_SIGNED_TOKEN_PARAM,
+  )
+
+  const token = useMemo(() => {
+    if (twoFactorAuthResetTokenParam) {
+      return hexDecodeAndDeserialize(
+        twoFactorAuthResetTokenParam,
+      ) as TwoFactorAuthResetToken
+    }
+    return undefined
+  }, [twoFactorAuthResetTokenParam])
+
+  // To disable 2FA, we must send both the token in the search param and a fresh twoFaToken from a login attempt
+  const [twoFaRequiredToken, setTwoFaRequiredToken] = useState<
+    Pick<TwoFactorAuthErrorResponse, 'twoFaToken' | 'userId'> | undefined
+  >()
+
+  const promptForSignIn = token && !twoFaRequiredToken
+  const promptConfirmDisable2FA = token && twoFaRequiredToken
+
+  const {
+    mutate: disableTwoFactorAuth,
+    isSuccess,
+    isPending,
+    error,
+  } = SynapseQueries.useDisableTwoFactorAuthWithResetToken({
+    onSuccess: () => {
+      displayToast('2FA has been successfully disabled on your account.')
+      history.push('/authenticated/myaccount')
+      // We must refresh the session to clear previous 2FA error responses and allow the user to log in.
+      refreshSession()
```

Comment:
After the user successfully disables 2FA, just send them to the account settings page, where they will have to sign in.

The user got here by opening a link in their email, so we don't have to send them back to an arbitrary page where they decided to sign in like we do for most other login scenarios.